plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/check/Declarations.kt

```diff
@@ -36,34 +38,39 @@ import org.jetbrains.kotlin.resolve.checkers.DeclarationCheckerContext
  * - whether the body satisfy all the pre-conditions in calls,
  * - whether the post-condition really holds.
  */
-internal fun SolverState.checkTopLevelDeclaration(
-  constraints: DeclarationConstraints?,
+internal fun SolverState.checkTopLevelDeclarationWithBody(
   context: DeclarationCheckerContext,
   descriptor: DeclarationDescriptor,
-  resultVarName: String,
   declaration: KtDeclaration
 ): ContSeq<Return> {
-  val maybeBody = declaration.stableBody()
-  return doOnlyWhenNotNull(maybeBody, NoReturn) { body ->
-    continuationBracket.map {
-      // introduce non-nullability of parameters
-      if (descriptor is CallableDescriptor) {
-        descriptor.allParameters.forEach { param ->
-          if (!param.type.isMarkedNullable) {
-            val paramName = param.name.asString()
-            val notNullFormula = solver.isNotNull(solver.makeObjectVariable(paramName))
-            addConstraint(NamedConstraint("$paramName is not null", notNullFormula))
-          }
+  // bring the constraints in (if there are any)
+  val constraints = constraintsFromSolverState(descriptor)
+  // choose a good name for the result
+  // should we change it for 'val' declarations?
```

Comment:
the current `$result` name is safe because it's not a kotlin identifier, but we can change it to whatever makes sense